index.js

```diff
@@ -50,7 +52,12 @@ module.exports = (nextConfig = {}) => ({
     config.entry = async () => {
       const entries = await originalEntry();
       if (entries['main.js'] && !dontAutoRegisterSw) {
-        entries['main.js'].unshift(require.resolve('./register-sw.js'));
+        let content = await readFile(require.resolve('./register-sw.js'), 'utf8');
+        content = content.replace('{REGISTER_SW_PREFIX}', registerSwPrefix);
+
+        await writeFile(join(__dirname, 'register-sw-compiled.js'), content, 'utf8');
```

Comment:
not certain about this block because service workers _must_ be served on the origin domain they're being used on.

So, the service worker on jackhanford.com would never register if it was being served on cdn.foobar.com.

https://github.com/w3c/ServiceWorker/issues/940

Reply:
ahh, disregard ^ if this is for serving the service worker on a different path on the same domain. ie `jackhanford.com/foo/bar/service-worker.js`